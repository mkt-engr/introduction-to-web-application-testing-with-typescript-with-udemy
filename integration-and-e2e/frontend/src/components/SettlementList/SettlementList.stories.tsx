import { Settlement } from "../../type";
import SettlementList from "./SettlementList";
import type { Meta, StoryObj } from "@storybook/react";

const meta = {
  title: "SettlementList",
  component: SettlementList,
} satisfies Meta<typeof SettlementList>;

export default meta;

type Story = StoryObj<typeof SettlementList>;

const settlements: Settlement[] = [
  { from: "Alice", to: "Bob", amount: 1000 },
  { from: "Bob", to: "Alice", amount: 2000 },
];

export const Default: Story = {
  args: {
    settlements,
  },
};
