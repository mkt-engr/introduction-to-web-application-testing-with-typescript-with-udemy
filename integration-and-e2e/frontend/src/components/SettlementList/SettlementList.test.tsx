import { Settlement } from "../../type";
import { render } from "@testing-library/react";
import SettlementList from "./SettlementList";
describe("SettlementList", () => {
  it("スナップショットテスト", () => {
    const settlements: Settlement[] = [
      { from: "Alice", to: "Bob", amount: 1000 },
      { from: "Bob", to: "Alice", amount: 2000 },
    ];

    const { container } = render(<SettlementList settlements={settlements} />);
    expect(container).toMatchSnapshot();
  });
});
