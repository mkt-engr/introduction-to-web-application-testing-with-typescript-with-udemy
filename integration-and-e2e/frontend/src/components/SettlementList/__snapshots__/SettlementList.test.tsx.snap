// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettlementList スナップショットテスト 1`] = `
<div>
  <div
    class="chakra-stack css-1anxrbq"
  >
    <p
      class="chakra-text css-x39wmu"
    >
      清算方法
    </p>
    <ul
      class="css-1s6xsev"
      role="list"
    >
      <li
        class="css-1v5z18m"
      >
        <div
          class="chakra-stack css-bot922"
        >
          <div
            class="css-0"
          >
            Alice
             → 
            Bob
          </div>
          <div
            class="css-0"
          >
            1000
            円
          </div>
        </div>
      </li>
      <li
        class="css-1v5z18m"
      >
        <div
          class="chakra-stack css-bot922"
        >
          <div
            class="css-0"
          >
            Bob
             → 
            Alice
          </div>
          <div
            class="css-0"
          >
            2000
            円
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
